### PR TITLE
[bootstrap agent] Remove unused trust quorum code

### PR DESCRIPTION
**Pull request by andrewjstone**
This removes a bunch of outdated trust quorum code and renames some types, functions, and messages for clarity. It should make it much easier to reason about bootstrap agent and sled agent startup.

### CODE REVIEW

File: sled-agent/src/bootstrap/client.rs [hunk 1]
```diff
@@ -67,55 +62,30 @@ pub enum Error {
...
-    }
-
-    pub(crate) fn addr(&self) -> SocketAddrV6 {
-        self.addr
+impl Client {
+    pub(crate) fn new(addr: SocketAddrV6, _log: Logger) -> Self {
+        Self { addr, _log }
     }
 
     pub(crate) async fn start_sled(
```
Comment (andrewjstone) [inline sled-agent/src/bootstrap/client.rs:81]: There really should be a comment about what this is doing here.

File: sled-agent/src/bootstrap/rss_handle.rs [hunk 2]
```diff
@@ -88,32 +81,14 @@ async fn initialize_sled_agent(
     log: &Logger,
```
Comment (andrewjstone) [inline sled-agent/src/bootstrap/rss_handle.rs:81]: There should be a header comment for this function

File: sled-agent/src/bootstrap/rss_handle.rs [hunk 3]
```diff
@@ -219,11 +189,7 @@ struct BootstrapAgentHandleReceiver {
 }
 
 impl BootstrapAgentHandleReceiver {
-    async fn await_local_request(
-        mut self,
-        log: &Logger,
-        sp: &Option<SpHandle>,
-    ) {
+    async fn await_local_request(mut self, log: &Logger) {
```
Comment (andrewjstone) [inline sled-agent/src/bootstrap/rss_handle.rs:192]: Header comment please

File: sled-agent/src/bootstrap/server.rs [hunk 4]
```diff
@@ -157,18 +132,12 @@ impl Server {
...
 struct Inner {
     listener: TcpListener,
-    sp: Option<SpHandle>,
-    trust_quorum: TrustQuorumMembership,
     bootstrap_agent: Arc<Agent>,
     log: Logger,
 }
 
 impl Inner {
     async fn start_sprockets(
```
Comment (andrewjstone) [inline sled-agent/src/bootstrap/server.rs:140]: This should probably get changed from `start_sprockets`, which it no longer even attempts, to `start_bootstrap_agent_listener` or something.

File: sled-agent/src/bootstrap/server.rs [hunk 5]
```diff
@@ -276,110 +179,19 @@ impl Inner {
...
-                warn!(log, "Share requested before we have one");
-                Err("Share request failed: share unavailable".to_string())
-            }
-        },
-    };
-
-    write_response(&mut stream, response).await
-}
-
 async fn serve_request_after_quorum_initialization(
```
Comment (andrewjstone) [inline sled-agent/src/bootstrap/server.rs:182]: We should change the name here to `handle_sled_init_request` or something

File: sled-agent/src/bootstrap/server.rs [hunk 6]
```diff
@@ -276,110 +179,19 @@ impl Inner {
...
-    .await
-    .map_err(|err| format!("Failed to establish sprockets session: {err}"))?;
+    let mut stream = Box::new(tokio::io::BufStream::new(stream));
 
     let response = match read_request(&mut stream).await? {
-        Request::SledAgentRequest(request, trust_quorum_share) => {
-            let trust_quorum_share =
-                trust_quorum_share.map(ShareDistribution::from);
-
+        Request::SledAgentRequest(request) => {
```
Comment (andrewjstone) [inline sled-agent/src/bootstrap/server.rs:190]: We should probably also change the name of this request to `StartSledAgentRequest`

File: sled-agent/src/bootstrap/rss_handle.rs [hunk 7]
```diff
@@ -238,27 +204,21 @@ impl BootstrapAgentHandleReceiver {
...
-                            sp,
-                        )
-                        .await
-                        .map_err(|err| {
-                            format!(
-                                "Failed to initialize sled agent at {}: {}",
-                                bootstrap_addr, err
-                            )
-                        })?;
+                        initialize_sled_agent(log, bootstrap_addr, &request)
```
Comment (andrewjstone) [inline sled-agent/src/bootstrap/rss_handle.rs:214]: Maybe rename this to `initialize_remote_sled_agent` ?

File: sled-agent/src/bootstrap/client.rs [hunk 8]
```diff
@@ -135,23 +108,14 @@ impl<'a> Client<'a> {
...
-            &self.log,
-        )
-        .await
-        .map_err(|err| Error::SprocketsSession(err.to_string()))?;
+        let mut stream = Box::new(tokio::io::BufStream::new(stream));
 
         // Build and serialize our request.
         let envelope = RequestEnvelope { version: version::V1, request };
 
         // "danger" note: `buf` contains a raw trust quorum share; we must not
```
Comment (jgallagher) [inline sled-agent/src/bootstrap/client.rs:116]: This comment can be removed too; there's no more "danger" to note.

**Review by jgallagher — COMMENTED** [review]
Changes all look fine to me. Big :+1: on giving this a test run on paris first though, it chunks out a lot of (hopefully unused but one can never be too careful) stuff.

File: sled-agent/src/bootstrap/client.rs (hunk 8 above)
Comment (andrewjstone) [reply sled-agent/src/bootstrap/client.rs]: done.

**Pull request merged**